vyper/context/definitions/bases.py

```diff
@@ -0,0 +1,484 @@
+from collections import (
+    OrderedDict,
+)
+from typing import (
+    Optional,
+    Tuple,
+    Union,
+)
+
+from vyper import (
+    ast as vy_ast,
+)
+from vyper.context import (
+    definitions,
+)
+from vyper.context.types import (
+    ValueType,
+    get_builtin_type,
+)
+from vyper.context.utils import (
+    compare_types,
+    validate_call_args,
+)
+from vyper.exceptions import (
+    ArrayIndexException,
+    CompilerPanic,
+    ConstancyViolation,
+    InvalidLiteral,
+    InvalidOperation,
+    NamespaceCollision,
+    StructureException,
+    TypeMismatch,
+    UnknownAttribute,
+    VyperException,
+)
+
+
+class BaseDefinition:
```

Comment:
Would this be a good candidate for Abstract base classes? https://docs.python.org/3/library/abc.html